include/rocksdb/rate_limiter.h

```diff
@@ -91,13 +92,16 @@ class RateLimiter {
 
   // Total # of requests that are pending for bytes in rate limiter
   // For convenience, this function is implemented by the RateLimiter returned
-  // by NewGenericRateLimiter but is not required by RocksDB. The default
-  // implementation indicates "not supported".
-  virtual int64_t GetTotalPendingRequests(
+  // by NewGenericRateLimiter but is not required by RocksDB.
+  //
+  // REQUIRED: total_pending_request != nullptr
+  virtual Status GetTotalPendingRequests(
+      int64_t* total_pending_requests,
```

Comment:
[Note]
I tried to order the output parameter after the input, similar to this comment https://github.com/facebook/rocksdb/pull/8722#discussion_r703682312 suggest. However, I also don't want to make `total_pending_requests` non-required :/ 

Any magic that I can use to get the good of both world?

Reply:
I think it's fine in this case. :)